File: reedb/tests/daemons/reedb.rb
# this is my_daemons.rb
require 'daemons'
require_relative '../../lib/reedb/utils/logger'
require_relative '../../lib/reedb/utils/utilities'

puts "Output 1"

# Initialize the app while we're not a daemon
Reedb::DaemonLogger.setup("/home/spacekookie/Desktop")
# server = TCPServer.open(2000)

puts "Init done"


# Become a daemon
Daemons.daemonize

Reedb::DaemonLogger.write("Daemon created")

# The server loop
loop do
	# puts "Loop!"
 #  client = server.accept
 #  params = JSON.parse(client.gets)
  Reedb::DaemonLogger.write("Still alive")
  sleep(1)
end

File: lib/private/reedb.cpp
/*
 * Reedb.cpp
 *
 *  Created on: 26 Jun 2015
 *      Author: spacekookie
 */

#include "../public/reedb.h"

Reedb::Reedb() {
	this->_core = new Core();
	this->_config = new Config();
	this->_vault = new Vault();
	this->_daemon = new Daemon();
}

Reedb::~Reedb() {

}

Core *Reedb::core() {
	return this->_core;
}

Config *Reedb::config() {
	return this->_config;
}

Vault *Reedb::vault() {
	return this->_vault;
}

Daemon *Reedb::daemon() {
	return this->_daemon;
}
